integrations/oci/oci-secrets-config-source/src/main/java/io/helidon/integrations/oci/secrets/configsource/OciSecretsConfigSourceProvider.java

```diff
@@ -0,0 +1,526 @@
+            return ABSENT_NODE_CONTENT;
+        }
+
+
+        /*
+         * Static methods.
+         */
+
+
+        private static Builder builder() {
+            return new Builder();
+        }
+
+        private static void completeAllSecretsTasks(Collection<? extends Callable<Void>> tasks, AutoCloseable secrets) {
+            RuntimeException re = null;
+            try (ExecutorService es = newVirtualThreadPerTaskExecutor()) {
+                for (Future<?> future : es.invokeAll(tasks)) {
+                    try {
+                        future.get();
+                    } catch (RuntimeException e) {
+                        if (re == null) {
+                            re = e;
+                        } else {
+                            re.addSuppressed(e);
+                        }
+                    } catch (Exception e) {
+                        if (e instanceof InterruptedException) {
+                            Thread.currentThread().interrupt();
+                        }
+                        if (re == null) {
+                            re = new IllegalStateException(e.getMessage(), e);
+                        } else {
+                            re.addSuppressed(e);
+                        }
+                    }
+                }
+            } catch (RuntimeException e) {
+                re = e;
+            } catch (Exception e) {
+                if (e instanceof InterruptedException) {
+                    Thread.currentThread().interrupt();
+                }
+                re = new IllegalStateException(e.getMessage(), e);
+            } finally {
+                try {
+                    secrets.close();
+                } catch (RuntimeException e) {
+                    if (re == null) {
+                        throw e;
+                    } else {
+                        re.addSuppressed(e);
+                    }
+                } catch (Exception e) {
+                    if (e instanceof InterruptedException) {
+                        Thread.currentThread().interrupt();
+                    }
+                    if (re == null) {
+                        throw new IllegalStateException(e.getMessage(), e);
+                    } else {
+                        re.addSuppressed(e);
+                    }
+                }
+                if (re != null) {
+                    throw re;
+                }
+            }
+        }
+
+        @SuppressWarnings("try")
+        private static Collection<? extends SecretSummary> secretSummaries(Supplier<? extends Vaults> vaultsSupplier,
+                                                                           ListSecretsRequest listSecretsRequest) {
+            try (Vaults v = vaultsSupplier.get()) {
+                return v.listSecrets(listSecretsRequest).getItems();
+            } catch (RuntimeException e) {
+                throw e;
+            } catch (Exception e) {
```

Comment:
Please replace all of these instances with catching the `InterruptedExceptinon` and handling it separately.

Reply:
OK. There will be duplicate code of course, since `Vaults#close()` throws `Exception` but I will make this change.